src/experiment/trials/callFunction.js

```diff
@@ -0,0 +1,16 @@
+import callFunction from "@jspsych/plugin-call-function";
+
+/**
+ * Trial to call the provided function
+ *
+ * @param {function} func will call this function
+ * @returns JS object as a trial
+ */
+export function buildCallFunctionTrial(func) {
+  return {
+    type: callFunction,
+    func: () => {
+      func;
+    },
```

Comment:
Little oddness of JavaScript here! Because we're passing a callback function to the property `func` we actually need to supply a callback function, and not just the _call_ to the function itself. There's a  couple ways that we can do this:


```suggestion
    // 1) This way is best
    func: func,
    // 2) Or this would work too
    func: () => {
      func()
    },
```

Notice how with option 2 we cannot pass any parameters to the function - even if we define them in `startProcedure.js`! Because of this I would go with option 1, and then in `startProcedure` you pass a callback function to the code we actually want to execute:

```js
procedure.push(buildCallFunctionTrial(() => { window.electronAPI.checkSerialPort() }));
```